### ISSUE TITLE
Should option parsing of ~/.sherpa.rc be more robust (i.e. not require all sections)?

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#490]
I created a simple Sherpa configuration file, intending to over-ride *only* the `options.plot_pkg` setting:

```
% cat ~/.sherpa.rc 
[options]
plot_pkg : pylab
```

Unfortunately our configuration-parsing code doesn't like this:

```
% python -c 'import sherpa.ui'
Traceback (most recent call last):
  File "/opt/ciao-4.10/ots/lib/python3.5/configparser.py", line 1135, in _unify_values
    sectiondict = self._sections[section]
KeyError: 'statistics'

During handling of the above exception, another exception occurred (I'm using the CIAOX branch here but master branch should perform very similarly):

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/ciao-4.10/lib/python3.5/site-packages/sherpa-4.10.0+28.g6692c01-py3.5-linux-x86_64.egg/sherpa/ui/__init__.py", line 20, in <module>
    import sherpa.all
  File "/opt/ciao-4.10/lib/python3.5/site-packages/sherpa-4.10.0+28.g6692c01-py3.5-linux-x86_64.egg/sherpa/all.py", line 37, in <module>
    from sherpa.fit import *
  File "/opt/ciao-4.10/lib/python3.5/site-packages/sherpa-4.10.0+28.g6692c01-py3.5-linux-x86_64.egg/sherpa/fit.py", line 38, in <module>
    from sherpa.stats import Chi2, Chi2Gehrels, Cash, CStat, Chi2ModVar, \
  File "/opt/ciao-4.10/lib/python3.5/site-packages/sherpa-4.10.0+28.g6692c01-py3.5-linux-x86_64.egg/sherpa/stats/__init__.py", line 48, in <module>
    truncation_flag = config.get('statistics', 'truncate').upper()
  File "/opt/ciao-4.10/ots/lib/python3.5/configparser.py", line 778, in get
    d = self._unify_values(section, vars)
  File "/opt/ciao-4.10/ots/lib/python3.5/configparser.py", line 1138, in _unify_values
    raise NoSectionError(section)
configparser.NoSectionError: No section: 'statistics'
```

Should we allow config files which only define a subset of values (this would make them more "future proof", in the sense that if we add a new configuration option/section then old `.sherpa.rc` files should continue to work)?

**Comment by DougBurke** [comment]
With python 3.5 the `ConfigParser` object supports a fallback/default value, which makes this easy to do. I don't think it's available with Python 2.7, so would need a slightly-different approach such as https://github.com/sherpa/sherpa/blob/master/sherpa/astro/io/__init__.py#L65

**Comment by DougBurke** [comment]
Closed by #684
